src/main/scala/zio/prelude/package.scala

```diff
@@ -73,6 +76,18 @@ package object prelude
     type Semilattice[A]        = Commutative[A] with Idempotent[A]
     type BoundedSemilattice[A] = Semilattice[A] with Identity[A]
 
+    type Semiring[A] =
+      AnnihilatingZero[A, CommutativeMonoid, Identity] with DistributiveMultiply[A, CommutativeMonoid, Identity]
+    type Ring[A]     =
+      AnnihilatingZero[A, AbelianGroup, Identity]
+        with DistributiveMultiply[A, AbelianGroup, Identity]
+        with SubtractShape[A, AbelianGroup, Identity]
+    type Field[A]    =
```

Comment:
These "proof of concepts" are useful and we should definitely include them.

Reply:
:+1: